### ISSUE TITLE
chunkserver: allow graceful recovery from read-only volume

### ISSUE THREAD

**Issue by onlyjob** [lizardfs/lizardfs#348]
One HDD began to malfunction and ext4 switched to read-only mode due to errors.
CGI marked volume as "damaged". I've noticed that and marked volume for removal since most files are still readable. However chunkserver started to log endless errors like the following:

    delete_chunk: file:/var/lib/lizardfs/00/chunks77/chunk_000000000077B28E_00000001.liz - unlink error: EROFS (Read-only file system)

From recovery prospective it might be useful if LizardFS avoid writing to volume that is marked for removal.

Maybe chunkserver could detect read-only mount and stop trying to modify volume?

Would it be useful to recognise another state, "read-only" to allow recovery from damaged but still (partially) readable volume?

**Comment by Zorlin** [comment]
I believe marking a disk for removal should reduce writing activity to
almost nothing. Is that not the case for you?

**Comment by onlyjob** [comment]
LizardFS do not create new chunks on volume marked for removal but it is actively deleting chunks. On read-only file system it produces flood of errors.

**Comment by biocyberman** [comment]
CGIserv reporting of missing chunks lead me here. I agree that chunk disk read-only mode is needed. It gives a clear signal to the administrator that the data on that disk is frozen and safe until further intervention.

**Comment by blink69** [comment]
On new versions LizardFS don't remove chunks from volume marked for removal.

**Comment by onlyjob** [comment]
> On new versions LizardFS don't remove chunks from volume marked for removal.

This is not true: 3.10.2 removes chunks from volumes marked for removal just like all releases prior to 3.10.2... Re-opening...
